clang/lib/DPCT/HeaderTypes.inc

```diff
@@ -0,0 +1,69 @@
+//===--------------- HeaderTypes.inc --------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef HEADER
+#define HEADER(Name, Spelling)
+#endif
+
+#ifndef STD_HEADER
+#define STD_HEADER(Name, Spelling) HEADER(Name, Spelling)
+#endif
+
+#ifndef ONEDPL_HEADER
+#define ONEDPL_HEADER(Name, Spelling) HEADER(DPL_ ## Name, Spelling)
+#endif
+
+#ifndef MKL_HEADER
+#define MKL_HEADER(Name, Spelling) HEADER(MKL_ ## Name, Spelling)
+#endif
+
+#ifndef DPCT_HEADER
+#define DPCT_HEADER(Name, Spelling) HEADER(DPCT_ ## Name, Spelling)
+#endif
+
+STD_HEADER(SYCL, "<sycl/sycl.hpp>")
+STD_HEADER(Math, "<cmath>")
+STD_HEADER(Algorithm, "<algorithm>")
+STD_HEADER(Complex, "<complex>")
+STD_HEADER(Thread, "<thread>")
+STD_HEADER(Future, "<future>")
+STD_HEADER(Time, "<time.h>")
+STD_HEADER(Stdlib, "<cstdlib>")
+STD_HEADER(Chrono, "<chrono>")
+STD_HEADER(Limits, "<limits>")
+STD_HEADER(Numeric, "<numeric>")
+STD_HEADER(Functional, "<functional>")
+#ifdef _WIN32
+STD_HEADER(DL, "<libloaderapi.h>")
+#else
+STD_HEADER(DL, "<dlfcn.h>")
+#endif
+
+ONEDPL_HEADER(Algorithm, "<oneapi/dpl/algorithm>")
+ONEDPL_HEADER(Execution, "<oneapi/dpl/execution>")
+ONEDPL_HEADER(Iterator, "<oneapi/dpl/iterator>")
+
+MKL_HEADER(Mkl, "<oneapi/mkl.hpp>")
+MKL_HEADER(RNG, "<oneapi/mkl/rng/device.hpp>")
```

Comment:
Can we include both <oneapi/mkl.hpp> and <oneapi/mkl/rng/device.hpp> for HT_MKL_RNG?

Reply:
done